dispatcher.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"io"
+	"log"
+	"net"
+)
+
+func game_tx(gameOutChannel chan InFrame, gameConn net.Conn) {
+	for {
+		log.Println("game_tx")
+		// Take the outbound frame to the game
+		frame := <-gameOutChannel
+
+		// Forward it to the game server
+		numWritten, err := gameConn.Write(frame.data) // TODO: Convert the entire frame to []byte and send it all
+		// TODO: Or perhaps these frames should be a different type altogether
+		if err != nil || numWritten < len(frame.data) { // See above TODO for len check
+			log.Println("Error writing to the game: ", err)
+			continue
+		}
+	}
+}
+
+func game_rx(gameInChannel chan OutFrame, gameConn net.Conn) {
+	const (
+		bufferSize = 4096
+		chunkSize  = 1024
+	)
+
+	for {
+		log.Println("game_rx")
+		buffer := make([]byte, 0, bufferSize)
+		chunk := make([]byte, chunkSize)
+
+		for { // Read the TCP response by chunks
+			numRead, err := gameConn.Read(chunk)
+
+			if err != nil {
```

Comment:
embedded comparison

Reply:
Not sure I understand? `break` happens for every error, but the `log` happens for a specific error.